src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -21,6 +21,19 @@ public SagaPersistenceBehavior(ISagaPersister persister, ICancelDeferredMessages
 
         public async Task Invoke(IInvokeHandlerContext context, Func<IInvokeHandlerContext, Task> next)
         {
+            var isTimeoutMessage = IsTimeoutMessage(context.Headers);
+            var isTimeoutHandler = context.MessageHandler.IsTimeoutHandler;
+
+            if (isTimeoutHandler && !isTimeoutMessage)
+            {
+                return;
+            }
+
+            if (!isTimeoutHandler && isTimeoutMessage)
+            {
+                return;
```

Comment:
No, we check that [here](https://github.com/Particular/NServiceBus/blob/18e4f48f34553206fc9ae9697ac12d93c19d6659/src/NServiceBus.Core/Sagas/Saga.cs#L55)